service_container/tags.rst

```diff
@@ -564,3 +564,242 @@ application handlers.
                 ->addTag('app.handler', ['priority' => 20]);
 
     Note that any other custom attributes will be ignored by this feature.
+
+
+Tagged Services Collection with Index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+What if you want to retrieve a specific service within the injected collection previously?
+The solution is to use the ``index_by`` and ``default_index_method`` options of the argument
+in combination with ``!tagged``.
+
+In the following example, all services tagged with ``app.handler`` are passed as
+first constructor argument to the ``App\Handler\HandlerCollection`` service, like previously,
+but we can now access to a specific injected service :
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            App\Handler\One:
+                tags:
+                    - { name: 'app.handler', index_attribute_name: 'handler_one' }
+
+            App\Handler\Two:
+                tags:
+                    - { name: 'app.handler', index_attribute_name: 'handler_two' }
+
+            App\HandlerCollection:
+                # inject all services tagged with app.handler as first argument
+                arguments: [!tagged { tag: 'app.handler', index_by: 'index_attribute_name' }]
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="App\Handler\One">
+                    <tag name="app.handler" index_attribute_name="handler_one" />
+                </service>
+
+                <service id="App\Handler\Two">
+                    <tag name="app.handler" index_attribute_name="handler_two" />
+                </service>
+
+                <service id="App\HandlerCollection">
+                    <!-- inject all services tagged with app.handler as first argument -->
+                    <argument type="tagged" tag="app.handler" index-by="index_attribute_name"/>
```

Comment:
```suggestion
                    <argument type="tagged" tag="app.handler" index-by="index_attribute_name" />
```

Not sure about this one, but on ahead tagged service an extra space was added before ending the tag.